content/tokio/topics/tracing.md

```diff
@@ -0,0 +1,249 @@
+    // Start configuring a `fmt` subscriber
+    let subscriber = tracing_subscriber::fmt()
+        // Use a more compact, abbreviated log format
+        .compact()
+        // Display source code file paths
+        .with_file(true)
+        // Display source code line numbers
+        .with_line_number(true)
+        // Display the thread ID an event was recorded on
+        .with_thread_ids(true)
+        // Don't display the event's target (module path)
+        .with_target(false)
+        // Build the subscriber
+        .finish();
+```
+
+For details on the available configuration options, see [the
+`tracing_subscriber::fmt` documentation][fmt-cfg].
+
+
+In addition to the [`FmtSubscriber`] type from [`tracing-subscriber`], other
+`Subscriber`s can implement their own ways of recording `tracing` data. This
+includes alternative output formats, analysis and aggregation, and integration
+with other systems such as distributed tracing or log aggregation services. A
+number of crates provide additional `Subscriber` implementations that may be of
+interest. See [here][related-crates] for an (incomplete) list of additional
+`Subscriber` implementations.
+
+Finally, in some cases, it may be useful to combine multiple different ways of
+recording traces together to build a single `Subscriber` that implements
+multiple behaviors. For this purpose, the `tracing-subscriber` crate provides a
+[`Layer`] trait that represents a component that may be composed together with
+other `Layer`s to form a `Subscriber`. See [here][`Layer`] for details on using
+`Layer`s.
+
+[fmt-cfg]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html#configuration
+[related-crates]: https://docs.rs/tracing/latest/tracing/index.html#related-crates
+[`Layer`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/index.html
+
+# Emitting Spans and Events
+
+The easiest way to emit spans is with the [`instrument`] proc-macro annotation
+provided by [`tracing`], which re-writes the bodies of functions to emit spans
+each time they are invoked.
+
+[`instrument`]: https://docs.rs/tracing/latest/tracing/attr.instrument.html
+
+For instance, to trace the method in `mini-redis-server` that handles each
+connection:
+
+```rust
+# struct Handler {
+#     connection: tokio::net::TcpStream,
+# }
+# pub type Error = Box<dyn std::error::Error + Send + Sync>;
+# pub type Result<T> = std::result::Result<T, Error>;
+use tracing::instrument;
+
+impl Handler {
+    /// Process a single connection.
+    #[instrument(
+        level = "info",
+        name = "Handler::run",
+        skip(self),
+        fields(
+            // `%` serializes the peer IP addr with `Display`
+            peer_addr = %self.connection.peer_addr().unwrap()
+        ),
+    )]
+    async fn run(&mut self) -> mini_redis::Result<()> {
+# /*
+        ...
+# */ Ok::<_, _>(())
+    }
+}
+```
+
+`mini-redis-server` will now emit a `tracing` Span for each incoming connection
+that:
```

Comment:
(This could/should probably just be a follow up PR)
It would be nice to have a sample of actual the log line in here